src/meshlink.c

```diff
@@ -2271,19 +1958,10 @@ bool meshlink_set_port(meshlink_handle_t *mesh, int port) {
 	}
 
 	close_network_connections(mesh);
-	exit_configuration(&mesh->config);
-
-	char portstr[10];
-	snprintf(portstr, sizeof(portstr), "%d", port);
-	portstr[sizeof(portstr) - 1] = 0;
-
-	modify_config_file(mesh, mesh->name, "Port", portstr, true);
 
-	init_configuration(&mesh->config);
+	// TODO: write meshlink.conf again
 
-	if(!read_server_config(mesh)) {
-		meshlink_errno = MESHLINK_ESTORAGE;
-	} else if(!setup_network(mesh)) {
+	if(!setup_network(mesh)) {
```

Comment:
I think here we need to create a node for ourself i.e `mesh->self` as it has been freed while `close_network_connections` before `setup_network` or inside `setup_network` before we call `setup_myself` where we use our node handle

Reply:
Yes, I've added code to recreate `mesh->self` before calling `setup_network()`. Maybe it would be even better if we can prevent `mesh->self` from being freed.